lib/hotel_booker.rb

```diff
@@ -0,0 +1,139 @@
+require 'date'
+require_relative 'room'
+require_relative 'reservation'
+require_relative 'date_range'
+
+module Hotel
+  class HotelBooker
+    attr_accessor :rooms, :reservations, :unreserved_block, :reserved_block
+    NUM_ROOMS = 20
+    BLOCK_MAX = 5
+
+    def initialize()
+      @unreserved_block = []
+      @reserved_block = []
+      @reservations = []
+      @rooms = []
+
+      NUM_ROOMS.times do |i|
+        @rooms << Hotel::Room.new(i+1)
+      end
+    end
+
+
+    # arguments for check_in and check_out are Strings
+    def make_reservation(id, check_in, check_out)
+      check_in = Date.parse(check_in)
+      check_out = Date.parse(check_out)
+      date_range = range(check_in, check_out)
+      reservation = Hotel::Reservation.new(id, date_range)
```

Comment:
It might give you less "overusing the word range"-anxiety to in-line this:
```ruby
reservation = Hotel::Reservation.new(id, range(check_in, check_out))
```